components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java

```diff
@@ -375,9 +375,21 @@ public abstract List<SpTemplate> getAllApplicationTemplateInfo(String tenantDoma
      * Get configured authenticators of an application.
      *
      * @param applicationID ID of an application.
+     * @param tenantDomain  Tenant domain.
      * @return list of configured authenticators.
      * @throws IdentityApplicationManagementException If error occurs in retrieving configured authenticators.
      */
+    public abstract AuthenticationStep[] getConfiguredAuthenticators(String applicationID, String tenantDomain)
+            throws IdentityApplicationManagementException;
+
+    /**
+     * Get configured authenticators of an application.
+     *
+     * @param applicationID ID of an application.
+     * @return list of configured authenticators.
+     * @throws IdentityApplicationManagementException If error occurs in retrieving configured authenticators.
+     */
+    @Deprecated
```

Comment:
When adding deprecated annotation, mention what to use instead of this and from which version this will be deprecated.